client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/HashBasedShuffleWriter.java

```diff
@@ -216,22 +233,22 @@ private void fastColumnarWrite0(scala.collection.Iterator iterator) throws IOExc
 
       if (rssBatchBuilders[partitionId] == null) {
         RssBatchBuilder columnBuilders;
-        if (conf.columnarShuffleCodeGenEnabled() && !conf.columnarShuffleDictionaryEnabled()) {
+        if (columnarShuffleCodeGenEnabled && !columnarShuffleDictionaryEnabled) {
           columnBuilders =
-              new RssColumnarBatchCodeGenBuild().create(schema, conf.columnarShuffleBatchSize());
+              new RssColumnarBatchCodeGenBuild().create(schema, columnarShuffleBatchSize);
         } else {
           columnBuilders =
               new RssColumnarBatchBuilder(
                   schema,
-                  conf.columnarShuffleBatchSize(),
-                  conf.columnarShuffleDictionaryMaxFactor(),
-                  conf.columnarShuffleDictionaryEnabled());
+                  columnarShuffleBatchSize,
+                  columnarShuffleDictionaryMaxFactor,
+                  columnarShuffleDictionaryEnabled);
         }
         columnBuilders.newBuilders();
         rssBatchBuilders[partitionId] = columnBuilders;
       }
       rssBatchBuilders[partitionId].writeRow(row);
-      if (rssBatchBuilders[partitionId].getRowCnt() >= conf.columnarShuffleBatchSize()) {
+      if (rssBatchBuilders[partitionId].getRowCnt() >= columnarShuffleBatchSize) {
```

Comment:
Thanks @cfmcgrady for optimizing this! It's surprising that it improves so much. I think it's because of this line since it's called for each record.

Reply:
yeah, here is the primary performance bottleneck before.

Reply:
> yeah, here is the primary performance bottleneck before.

BTW, in your former PR https://github.com/apache/incubator-celeborn/pull/1585#issuecomment-1589164128, I thought the improvement can also be applied to hash writer, WDYT?

Reply:
sure, we can do the same refactor for the hash-based writer too, I'll raise a PR later.